[NU-2301] Fix for ClassCastException when using json templates with m…

arkadius opened the pull request:
…ore complex types: passing ValidationContext to expression compiled for lazy parameters as well

## Describe your changes

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

arkadius committed:
[NU-2301] Fix for ClassCastException when using json templates with more complex types: passing ValidationContext to expression compiled for lazy parameters as well

arkadius committed:
real fix

arkadius committed:
parseWithoutContextValidation removal

arkadius committed:
More types + class extraction settings

arkadius committed:
SingleInputNodeInputValidationContext explicitly passed everywhere

arkadius committed:
ValidationContext for Lazy parameters from compilation process

arkadius committed:
module not opens to unnamed module fix

arkadius committed:
module not opens to unnamed module fix #2: added add-opens

raphaelsolarski reviewed (approved):
LGTM.
Changelog entry?

arkadius committed:
- Base on touk/flink image in `WithFlinkContainers` instead of adding jar artifact
- Custom TypeInformation provided by detection mechanism to avoid ClassNotFoundException
- Changelog entry

arkadius committed:
Merge remote-tracking branch 'origin/staging' into NU-2301-using-validation-context-in-lazyparam

arkadius committed:
fix after merge

arkadius committed:
ZoneOffset fix
